packages/spl-token-bonding/src/index.ts

```diff
@@ -1471,29 +1435,46 @@ export class SplTokenBonding extends AnchorSdk<SplTokenBondingIDL> {
       // @ts-ignore
       buyWithBase,
     };
-    const accounts = {
-      accounts: {
-        tokenBonding,
-        // @ts-ignore
-        curve: tokenBondingAcct.curve,
-        baseMint: tokenBondingAcct.baseMint,
-        targetMint: tokenBondingAcct.targetMint,
-        baseStorage: tokenBondingAcct.baseStorage,
-        buyBaseRoyalties: tokenBondingAcct.buyBaseRoyalties,
-        buyTargetRoyalties: tokenBondingAcct.buyTargetRoyalties,
-        source,
-        sourceAuthority,
-        destination,
-        tokenProgram: TOKEN_PROGRAM_ID,
-        clock: SYSVAR_CLOCK_PUBKEY,
-      },
+    const common = {
+      tokenBonding,
+      // @ts-ignore
+      curve: tokenBondingAcct.curve,
+      baseMint: tokenBondingAcct.baseMint,
+      targetMint: tokenBondingAcct.targetMint,
+      baseStorage: tokenBondingAcct.baseStorage,
+      buyBaseRoyalties: tokenBondingAcct.buyBaseRoyalties,
+      buyTargetRoyalties: tokenBondingAcct.buyTargetRoyalties,
+      tokenProgram: TOKEN_PROGRAM_ID,
+      clock: SYSVAR_CLOCK_PUBKEY,
+      destination
     };
-    instructions.push(await this.instruction.buyV0(args, accounts));
-    instructions.push(...lastInstructions);
+
+    if (isNative) {
+      instructions.push(await this.instruction.buyNativeV0(args, { 
+        accounts: {
+          common,
+          state: state.publicKey,
+          wrappedSolMint: state.wrappedSolMint,
+          mintAuthority: (await SplTokenBonding.wrappedSolMintAuthorityKey(this.programId))[0],
```

Comment:
Would it make sense to just have 
```javascript
SplTokenBonding.wrappedSolMintAuthorityKey
```
return the item at the ```[0]``` index? is there ever a time we dont do that?